src/widgets/MultiSelectList.js

```diff
@@ -0,0 +1,235 @@
+        regexFilter.test(optionItem[secondaryFilterProperty])
+    );
+  };
+
+  /**
+   * Delegator of filtering process. Check if the object is a realm
+   * object (has the filtered member) or if it is an array. Otherwise,
+   * return an empty list to display.
+   */
+  getData = () => {
+    const { options, primaryFilterProperty, queryString } = this.props;
+    if (options && options.filtered && queryString) return this.filterResultData(options);
+    if (Array.isArray(options) && primaryFilterProperty) return this.filterArrayData(options);
+    return [];
+  };
+
+  /**
+   * Render single item of the FlatList
+   *
+   * @prop  {object}     data   Single item data.
+   *
+   * @returns {component} ResultRow
+   *
+   */
+  renderItem = data => {
+    const { renderLeftText, renderRightText, showCheckIcon } = this.props;
+    return (
+      <ResultRow
+        data={data}
+        onPress={() => this.onSelect(data)}
+        isSelected={() => this.isSelected(data)}
+        renderLeftText={renderLeftText}
+        renderRightText={renderRightText}
+        showCheckIcon={showCheckIcon}
+      />
+    );
+  };
+
+  /**
+   * On select FlatList item update selected state
+   *
+   * @prop  {object}     data   Single item data.
+   */
+  onSelect = data =>
+    this.setState(prevState =>
+      prevState.selected.indexOf(data.item.id) === -1
+        ? { selected: [...prevState.selected, data.item.id] }
+        : { selected: prevState.selected.filter(i => i !== data.item.id) }
+    );
+
+  /**
+   * Check if the single item is selected or not
+   *
+   * @prop  {object}     data   Single item data.
+   *
+   * @returns {bool}    selected state of the item
+   */
+  isSelected = data => {
+    const { selected } = this.state;
+    return !(selected.indexOf(data.item.id) === -1);
+  };
+
+  render() {
+    const { placeholderText, onConfirmSelections } = this.props;
+    const { selected } = this.state;
+    const data = this.getData();
+
+    return (
+      <View style={localStyles.container}>
+        <SearchBar
+          autoCapitalize="none"
+          autoCorrect={false}
+          autoFocus
+          color="white"
+          onChange={text => this.setState({ queryText: text })}
+          placeholder={placeholderText}
+          placeholderTextColor="white"
+          style={[localStyles.text, localStyles.searchBar]}
+        />
+        {data.length > 0 && (
+          <FlatList
+            data={data}
+            keyExtractor={item => item.id || item.name}
+            renderItem={item => this.renderItem(item)}
+            keyboardShouldPersistTaps="always"
+            style={localStyles.resultList}
+            extraData={selected}
+          />
+        )}
+        <View style={localStyles.contentContainer}>
+          <View style={[localStyles.buttonContainer]}>
+            <OnePressButton
+              style={[globalStyles.button, localStyles.confirmButton]}
```

Comment:
these arrays of styles cause needless re-renders as the references change, just like a function. There's really no point for a few - for example this one could just all be defined as a local style.